ads/inabox/inabox-messaging-host.js

```diff
@@ -224,12 +225,24 @@ export class InaboxMessagingHost {
       return this.iframeMap_[sentinel];
     }
 
-    // Walk up on the window tree until find host's direct child window
-    while (source.parent !== this.win_ && source !== this.win_.top) {
-      source = source.parent;
+    // Walk up on the window tree and verify that there is no cross-domain
+    // frame in between the source of the postMessage and the host doc.
+    // If there is, the host doc will not be able to accurately measure
+    // the creative's positions in the page, so return null.
+    // Limit to 10 iterations.
+    let tempWin = source.parent;
```

Comment:
Let's only do this if the frame owning the message exists in the array.  This will potentially save CPU cycles.

Reply:
done